### PR TITLE
OpenRouter Integration

### PR DESCRIPTION
**Pull request by Anindyadeep**
This PR:

- Solves issue #20

### CODE REVIEW

**Comment by Anindyadeep** [comment]
I will merge this after PR #47 get's merged and also don't I need to change the slug here? I mean there are still some identical

File: prem_utils/models.json [hunk 1]
```diff
@@ -624,6 +624,131 @@
...
+                    "model_type": "text2text",
+                    "context_tokens": 8000
+                },
+                {
+                    "slug": "nousresearch/nous-hermes-llama2-13b",
+                    "model_type": "text2text",
+                    "context_tokens": 4096
+                },
+                {
+                    "slug": "meta-llama/codellama-34b-instruct",
```
Comment (filopedraz) [inline prem_utils/models.json:691]: No slug conflicts here?

**Review by filopedraz — CHANGES_REQUESTED** [review]

File: prem_utils/connectors/openrouter.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        presence_penalty: float = 0,
+        seed: int | None = None,
+        stop: str | list[str] = None,
+        stream: bool = False,
+        temperature: float = 1,
+        top_p: float = 1,
+        tools: list[dict[str, Any]] = None,
+        tool_choice: dict = None,
+    ):
+        if "openrouter" in model:
```
Comment (filopedraz) [inline prem_utils/connectors/openrouter.py:28]: Now that you put `openrouter/` in front of each element, you don't need this if/else logic. 

```python
model = model.replace("openrouter/", "", 1)
```

Will already replace only the first occurence.

**Review by filopedraz — CHANGES_REQUESTED** [review]

File: prem_utils/connectors/openrouter.py (hunk 2 above)
Comment (Anindyadeep) [reply prem_utils/connectors/openrouter.py]: Done

**Review by filopedraz — APPROVED** [review]

**Pull request merged**